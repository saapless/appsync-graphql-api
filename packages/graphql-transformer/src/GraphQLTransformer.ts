export class GraphQLTransformer {
  constructor() {}

  public transform() {
    return "transformed";
  }
}
