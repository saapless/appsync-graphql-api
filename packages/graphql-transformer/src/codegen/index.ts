export * from "./ast";
export * from "./printer";
export * from "./document";
