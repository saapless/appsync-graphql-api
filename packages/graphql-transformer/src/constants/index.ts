export const ConnectionDirective = Object.freeze({
  HAS_ONE: "hasOne",
  HAS_MANY: "hasMany",
});

export const ModelDirective = Object.freeze({
  MODEL: "model",
});

export const UtilityDirective = Object.freeze({
  SERVER_ONLY: "serverOnly",
  CLIENT_ONLY: "clientOnly",
  READ_ONLY: "readOnly",
  WRITE_ONLY: "writeOnly",
  FILTER_ONLY: "filterOnly",
  INTERNAL: "internal",
});

export const ScalarType = Object.freeze({
  STRING: "String",
  INT: "Int",
  FLOAT: "Float",
  BOOLEAN: "Boolean",
  ID: "ID",
  LONG: "Long",
});

export const RelationType = Object.freeze({
  ONE_TO_ONE: "oneToOne",
  ONE_TO_MANY: "oneToMany",
  MANY_TO_MANY: "manyToMany",
});
