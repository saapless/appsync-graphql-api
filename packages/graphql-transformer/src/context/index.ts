export * from "./TransformerContext";
export type {
  DescriptorOperation,
  Key,
  KeyOperator,
  KeyValue,
  OperationActionType,
  ResolverDescriptor,
  ReturnType,
} from "./ResolverLoader";
