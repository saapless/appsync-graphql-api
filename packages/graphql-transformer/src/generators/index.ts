export { ResolverGeneratorBase } from "./ResolverGeneratorBase";
export { DynamoDbGenerator } from "./DynamoDBGenerator";
export { NoneGenerator } from "./NoneGenerator";
export { SchemaTypesGenerator } from "./SchemaTypesGenerator";
export { ResolverTypesGenerator } from "./ResolverTypesGenerator";
