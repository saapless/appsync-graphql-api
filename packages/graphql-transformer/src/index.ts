export * from "./transformer";
export { definitionFromSource } from "./utils";
