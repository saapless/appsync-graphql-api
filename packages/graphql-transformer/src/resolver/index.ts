export { ResolverBase } from "./ResolverBase";
export { FieldResolver } from "./FieldResolver";
export { FunctionResolver } from "./FunctionResolver";
