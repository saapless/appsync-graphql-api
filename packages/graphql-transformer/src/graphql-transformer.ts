import { GraphQLTransformer } from "./GraphQLTransformer";

export function createTransformer() {
  return new GraphQLTransformer();
}

export function executeTranformer() {}
