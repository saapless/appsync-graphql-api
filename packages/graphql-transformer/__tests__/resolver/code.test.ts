import { MemberExpression, NodeKind, printAST, tc } from "../../src/resolver";

describe("Resolver code compiler", () => {
  describe("the `tc` ast utility", () => {
    it.todo("print array expression");
    it.todo("print arrow function statement");
    it.todo("print call expression");
    it("print chain expression", () => {
      const ast = tc.chain("ctx.args.id");

      expect(ast).toEqual({
        _kind: NodeKind.MEMBER_EXPRESSION,
        object: {
          _kind: NodeKind.MEMBER_EXPRESSION,
          object: {
            _kind: NodeKind.IDENTIFIER,
            name: "ctx",
          },
          property: {
            _kind: NodeKind.IDENTIFIER,
            name: "args",
          },
          computed: false,
          optional: false,
        },
        property: {
          _kind: NodeKind.IDENTIFIER,
          name: "id",
        },
        computed: false,
        optional: false,
      } satisfies MemberExpression);
    });
    it.todo("print desctructuring expression");
    it.todo("print export statement");
    it.todo("print for statement");
    it("print function definition", () => {
      const ast = tc.func(
        "getTodo",
        [tc.ref("id")],
        tc.return(tc.obj(tc.prop("uuid", tc.ref("id"))))
      );
      expect(printAST(ast)).toMatchSnapshot();
    });
    it("print if statement", () => {
      const ast = tc.if(
        tc.eq(tc.ref("id"), tc.str("1")),
        tc.return(tc.obj(tc.prop("uuid", tc.ref("id")))),
        tc.return(tc.null())
      );

      expect(printAST(ast)).toMatchSnapshot();
    });
    it.todo("print import statement");
    it.todo("print logical expression");
    it.todo("print object expression");
    it.todo("print switch statement");
    it.todo("print ternary expression");
    it.todo("print type definition");
    it.todo("print type expression");
    it.todo("print unary expression");
    it.todo("print variable definition");
  });

  it.todo("print a code template document");
});
