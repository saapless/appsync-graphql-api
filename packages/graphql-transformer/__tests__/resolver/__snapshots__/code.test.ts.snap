// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resolver code compiler the \`tc\` ast utility print function definition 1`] = `
"function getTodo(id) {
  return { uuid: id }
}
"
`;

exports[`Resolver code compiler the \`tc\` ast utility print if statement 1`] = `
"if (id === 1) {
  return { uuid: id }
} else return null"
`;
